File: SpringBootApp/src/main/java/com/example/springbootapp/Data/DTO/ProductWithVariantAvailabilityDto.java
package com.example.springbootapp.Data.DTO;

import com.example.springbootapp.Data.Entities.Enums.Size;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class ProductWithVariantAvailabilityDto {
    private String id;
    private Size size;
    private Integer availability;
}

File: ClientUser/app/src/main/java/com/example/clientuser/model/dto/ProductWithVariantAvailabilityDto.kt
package com.example.clientuser.model.dto

import com.example.clientuser.model.enumerator.Size
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class ProductWithVariantAvailabilityDto(
    val id: String,
    val size: Size,
    val availability: Int
)
